stdlib/messaging/rabbitmq/src/main/java/org/ballerinalang/messaging/rabbitmq/RabbitMQTransactionContext.java

```diff
@@ -0,0 +1,137 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.ballerinalang.messaging.rabbitmq;
+
+import com.rabbitmq.client.Channel;
+import org.ballerinalang.bre.Context;
+import org.ballerinalang.model.values.BMap;
+import org.ballerinalang.model.values.BValue;
+import org.ballerinalang.util.transactions.BallerinaTransactionContext;
+import org.ballerinalang.util.transactions.TransactionLocalContext;
+import org.ballerinalang.util.transactions.TransactionResourceManager;
+
+import java.io.IOException;
+import java.util.Objects;
+
+import javax.transaction.xa.XAResource;
+
+/**
+ * Channel wrapper class to handle Ballerina transaction logic.
+ *
+ * @since 0.995
+ */
+public class RabbitMQTransactionContext implements BallerinaTransactionContext {
+    private Channel channel;
+    private final String connectorId;
+    private long deliveryTag = -1;
+    private boolean autoAck = true;
+
+
+    /**
+     * Sets the value of autoAck.
+     *
+     * @param autoAck Ack mode of the consumer.
+     */
+    public void setAutoAck(boolean autoAck) {
+        this.autoAck = autoAck;
+    }
+
+    /**
+     * Sets the delivery tag of the message.
+     *
+     * @param deliveryTag Delivery tag of the message.
+     */
+    public void setDeliveryTag(long deliveryTag) {
+        this.deliveryTag = deliveryTag;
+    }
+
+    /**
+     * Initializes the rabbitmq transaction context.
+     *
+     * @param channelObject RabbitMQ Channel object.
+     * @param connectorId   Connector Id.
+     */
+    public RabbitMQTransactionContext(BMap<String, BValue> channelObject, String connectorId) {
+        channel = (Channel) channelObject.getNativeData(RabbitMQConstants.CHANNEL_NATIVE_OBJECT);
+        this.connectorId = connectorId;
+    }
+
+    @Override
+    public void commit() {
+        try {
+            if (deliveryTag != -1 && !autoAck) {
+                channel.basicAck(deliveryTag, false);
+            }
+            channel.txCommit();
+        } catch (IOException exception) {
+            throw new RabbitMQConnectorException(RabbitMQConstants.COMMIT_FAILED
+                    + exception.getMessage(), exception);
+        }
+    }
+
+    @Override
+    public void rollback() {
+        try {
+            if (deliveryTag != -1 && !autoAck) {
+                channel.basicNack(deliveryTag, false, true);
+            }
+            channel.txRollback();
+        } catch (IOException exception) {
+            throw new RabbitMQConnectorException(RabbitMQConstants.ROLLBACK_FAILED
+                    + exception.getMessage(), exception);
+        }
+
+    }
+
+    @Override
+    public void close() {
+        // Do nothing
+    }
+
+    @Override
+    public XAResource getXAResource() {
+        // Not supported
```

Comment:
We should return a runtime exception here saying not supported.

Reply:
Resolved